scanpy/tests/test_normalization.py

```diff
@@ -78,27 +78,19 @@ def test_normalize_pearson_residuals_inputchecks(sparsity_func, dtype):
         adata_noninteger = adata.copy()
         x, y = np.nonzero(adata_noninteger.X)
         adata_noninteger.X[x[0], y[0]] = 0.5
-        nonint_warn_msg = "`normalize_pearson_residuals()` expects raw count data, but non-integers were found."
 
-        # expecting 0 no-int warnings
-        with warnings.catch_warnings(record=True) as record:
+        with pytest.warns(UserWarning) as record:
```

Comment:
I spent way too much on this 😓 weirdly enough the return was always a list and not `UserWarning`. I used this function somewhere else and that time worked, don't know (not important, just a comment)